mne/viz/epochs.py

```diff
@@ -88,27 +90,109 @@ def plot_epochs_image(epochs, picks=None, sigma=0., vmin=None,
         and is added to the image. It is typically useful to display reaction
         times. Note that it is defined with respect to the order
         of epochs such that overlay_times[0] corresponds to epochs[0].
+    combine : None | str | callable
+        ...
+    groupby : None | str | dict
+        ...
 
     Returns
     -------
     figs : lists of matplotlib figures
         One figure per channel displayed.
     """
-    from scipy import ndimage
     units = _handle_default('units', units)
     scalings = _handle_default('scalings', scalings)
 
-    import matplotlib.pyplot as plt
+    picks = np.atleast_1d(picks)
+    if (fig is not None or axes is not None) and len(picks) > 1:
+        raise ValueError('Only single pick can be drawn to a figure.')
+
     if picks is None:
         picks = pick_types(epochs.info, meg=True, eeg=True, ref_meg=False,
                            exclude='bads')[:5]
 
     if set(units.keys()) != set(scalings.keys()):
         raise ValueError('Scalings and units must have the same keys.')
 
-    picks = np.atleast_1d(picks)
-    if (fig is not None or axes is not None) and len(picks) > 1:
-        raise ValueError('Only single pick can be drawn to a figure.')
+    ch_types = [channel_type(epochs.info, idx) for idx in picks]
+    for ch_type in ch_types:
+        if ch_type not in scalings:
+            # We know it's not in either scalings or units since keys match
+            raise KeyError('%s type not in scalings and units' % ch_type)
+
+    # First, we collect groupings of picks and types in two lists
+    # (all_picks, all_ch_types) -> groupby.
+    # Then, we construct a list of the corresponding data, names and evokeds
+    # (to_plot_list) -> combine.
+    # Then, we loop over this list and plot using _plot_epochs_image.
+
+    # groupby
+    if groupby is None:
+        all_picks, all_ch_types = [picks], ch_types   # fix!!!!
+    else:
+        if groupby == "type":
+            all_picks, all_ch_types = list(), list()
+            for this_type in set(ch_types):
+                these_picks = picks[np.array(ch_types) == this_type]
+                all_picks.append(these_picks)
+                all_ch_types.append(this_type)
+        if isinstance(groupby, dict):
+            if combine is None:
+                msg = "If `groupby` is a dict, `combine` must not be None"
+                raise ValueError(msg)
+            all_ch_types = list(groupby.keys())
+            all_picks = [groupby[ch_type] for ch_type in all_ch_types]
+
+    # combine/construct list for plotting
+    data = epochs.get_data()
+    to_plot_list = list()
+    if combine is None:
+        for pick, ch_type in zip(picks, all_ch_types):
+            this_data = data[:, pick, :].squeeze() #* scalings[ch_type] # (n_epochs, n_channels, n_times)
```

Comment:
The test fails because ``this_data`` is 1d in when using a single pick. I'm not sure you want to do this, though. It looks like you're averaging each channel separately.